### PR TITLE
remove mapbox-gl-supported module

### PR DESCRIPTION
**Pull request by zhangyiatmicrosoft**
## Launch Checklist

#2452 
Net saving of 2186 bytes on minimized script from 761259 down to 759073

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Document any changes to public APIs.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: src/ui/map.ts [hunk 1]
```diff
@@ -2709,7 +2709,13 @@ class Map extends Camera {
     }
 
     _setupPainter() {
-        const attributes = extend({}, supported.webGLContextAttributes, {
+        const webGLContextAttributes = {
```
Comment (HarelM) [inline src/ui/map.ts:2712]: This assumes we are using modern browsers, doesn't it?
Is there an easy why to check web gl supported features without using mapbox-gl-supported?

**Comment by HarelM** [comment]
I generally think this is a good direction, I don't like stale libraries, but we need to make sure we cover the relevant cases.

File: src/ui/map.ts (hunk 1 above)
Comment (IvanSanchez) [reply src/ui/map.ts:2712]: No, this just mimicks the behaviour of `@mapbox/mapbox-gl-supported`. No effort is made to detect support of `antialias`, `alpha`, `stencil` and `depth` (rather, see https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext#browser_compatibility - only Safari is a problem)

File: src/ui/map.ts [hunk 2]
```diff
@@ -2709,7 +2709,13 @@ class Map extends Camera {
...
+            antialias: false,
+            alpha: true,
+            stencil: true,
+            depth: true
+        };
+        const attributes = extend({}, webGLContextAttributes, {
```
Comment (IvanSanchez) [inline src/ui/map.ts:2718]: I'd rather get rid of `extend()` and use a single object literal, something like

```
const attributes = {
            alpha: true,
            stencil: true,
            depth: true,
            failIfMajorPerformanceCaveat: this._failIfMajorPerformanceCaveat,
            preserveDrawingBuffer: this._preserveDrawingBuffer,
            antialias: this._antialias ?? false
};
```

**Review by IvanSanchez — CHANGES_REQUESTED** [review]

File: src/ui/map.ts (hunk 2 above)
Comment (HarelM) [reply src/ui/map.ts:2718]: I don't like `extend` either, super hard to understand what you'll get back after using it...

File: src/ui/map.ts (hunk 1 above)
Comment (zhangyiatmicrosoft) [reply src/ui/map.ts:2712]: yes

File: src/ui/map.ts (hunk 2 above)
Comment (zhangyiatmicrosoft) [reply src/ui/map.ts:2718]: AGREE!

**Review by HarelM — APPROVED** [review]

**Comment by pramilk** [comment]
@HarelM, @zhangyiatmicrosoft 
Does this documention need to be updated?
https://maplibre.org/maplibre-gl-js-docs/api/properties/#supported

**Comment by HarelM** [comment]
I guess the example is either not relevant anymore or needs to be updated, I hope the supported will be removed automatically once the docs will be generated for version 3.
In any case, both issues should be addressed in the docs repo for now (until we move it here).

**Comment by ambientlight** [comment]
@HarelM, @zhangyiatmicrosoft: should we fork the `@mapbox/mapbox-gl-supported`? since per webgl2 by default we would need to check for the existence of `webgl2` I guess now?

**Pull request merged**